src/popup/compatMode.js

```diff
@@ -0,0 +1,49 @@
+browser.windows.getCurrent().then((windowInfo) => {
+  document.body.classList.toggle("basic", !windowInfo.incognito);
+  document.body.classList.toggle("private", windowInfo.incognito);
+});
+
+const getCurrentWindowActiveTab = () => {
+  return browser.tabs.query({currentWindow: true, active: true});
+};
+
+const button = document.getElementById("fixThisSite");
+button.addEventListener("click", (e) => {
+  getCurrentWindowActiveTab().then((tabList) => {
+    const activeTabID = tabList[0].id;
+    // Send message to feature.js to turn on compat mode.
+    browser.runtime.sendMessage({msg: "compat_mode", tabId: activeTabID});
+  });
+  window.close();
+});
+
+const anchors = document.querySelectorAll(".learn-more");
+for (const anchor of anchors) {
+  anchor.href = browser.runtime.getURL("./onboarding/index.html");
+}
+
+const extractHostname = (url) => {
```

Comment:
Can't you just use `new URL(url).hostname`?

Reply:
👍